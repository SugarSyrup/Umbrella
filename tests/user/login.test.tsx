import {render, screen} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Home from '@/pages/user/login';

describe('Login Page', () => {
    it('should have a username and a password field, also a submit button', () => {
        render(<Home />);

        const usernameField = screen.getByPlaceholderText(/Email/i);
        const passwordField = screen.getByPlaceholderText(/Password/i);
        const submitButton = screen.getByDisplayValue('Log in');
        
        expect(usernameField).toBeInTheDocument();
        expect(passwordField).toBeInTheDocument();
        expect(submitButton).toBeInTheDocument();
    });

    it('should allow the user to submit their credentials', () => {
        const submit = jest.fn();
        render(<Home submit={submit} />);

        const usernameField = screen.getByPlaceholderText(/Email/i);
        const passwordField = screen.getByPlaceholderText(/Password/i);
        const submitButton = screen.getByDisplayValue('Log in');

        userEvent.type(usernameField, "admin");
        userEvent.type(passwordField, "1q2w3e4r!");
        userEvent.click(submitButton);

        expect(submit).toHaveBeenCalledWith({
            username:"admin",
            password:"1q2w3e4r!",
        });
    })
})

//https://dev-yakuza.posstree.com/ko/react/nextjs/test/
//https://nextjs.org/docs/testing#setting-up-jest-with-the-rust-compiler


// error 해결 1


// Configuration error:
    
// Could not locate module @/pages/user/login mapped as:
// C:\WorkSpace\umbrella\src\$1.

// Please check your configuration for these entries:
// {
//   "moduleNameMapper": {
//     "/@\/(.*)$/": "C:\WorkSpace\umbrella\src\$1"
//   },
//   "resolver": undefined
// }

// https://github.com/vuejs/vue-jest/issues/40


//error 2

//  Test environment jest-enviroment-jsdom cannot be found. Make sure the testEnvironment configuration option points to an existing node module.

// or

// Implicit config resolution does not allow multiple configuration files.
// Either remove unused config files or select one explicitly with `--config`.        


//https://stackoverflow.com/questions/72013449/upgrading-jest-to-v29-error-test-environment-jest-environment-jsdom-cannot-be