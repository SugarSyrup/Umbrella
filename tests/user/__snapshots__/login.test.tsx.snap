// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders a login page 1`] = `
<div>
  <div
    class="mainContainer"
  >
    <main
      class="loginContainer"
    >
      <div
        class="textContainer"
      >
        <span
          class="heading"
        >
          Log in
        </span>
        <span
          class="smallTxt"
        >
          New to Umbrella? 
          <a
            href="/user/join"
          >
            Create an new account
          </a>
        </span>
      </div>
      <form
        class="loginForm"
      >
        <input
          name="username"
          placeholder="Email"
        />
        <p
          style="color: red;"
        />
        <input
          name="password"
          placeholder="Password"
          type="password"
        />
        <p
          style="color: red;"
        />
        <a
          href="/user/find-password"
        >
          Forgot Password?
        </a>
        <input
          class="submit"
          type="submit"
          value="Log in"
        />
      </form>
      <span
        class="hrr"
      >
        or
      </span>
      <div
        class="icons"
      >
        <div
          class="google"
        >
          <svg
            enable-background="new 0 0 48 48"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            version="1.1"
            viewBox="0 0 48 48"
            width="1em"
            x="0px"
            xmlns="http://www.w3.org/2000/svg"
            y="0px"
          >
            <path
              d="M43.611,20.083H42V20H24v8h11.303c-1.649,4.657-6.08,8-11.303,8c-6.627,0-12-5.373-12-12
	c0-6.627,5.373-12,12-12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657C34.046,6.053,29.268,4,24,4C12.955,4,4,12.955,4,24
	c0,11.045,8.955,20,20,20c11.045,0,20-8.955,20-20C44,22.659,43.862,21.35,43.611,20.083z"
              fill="#FFC107"
            />
            <path
              d="M6.306,14.691l6.571,4.819C14.655,15.108,18.961,12,24,12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657
	C34.046,6.053,29.268,4,24,4C16.318,4,9.656,8.337,6.306,14.691z"
              fill="#FF3D00"
            />
            <path
              d="M24,44c5.166,0,9.86-1.977,13.409-5.192l-6.19-5.238C29.211,35.091,26.715,36,24,36
	c-5.202,0-9.619-3.317-11.283-7.946l-6.522,5.025C9.505,39.556,16.227,44,24,44z"
              fill="#4CAF50"
            />
            <path
              d="M43.611,20.083H42V20H24v8h11.303c-0.792,2.237-2.231,4.166-4.087,5.571
	c0.001-0.001,0.002-0.001,0.003-0.002l6.19,5.238C36.971,39.205,44,34,44,24C44,22.659,43.862,21.35,43.611,20.083z"
              fill="#1976D2"
            />
          </svg>
        </div>
        <div
          class="kakao"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 3c5.799 0 10.5 3.664 10.5 8.185 0 4.52-4.701 8.184-10.5 8.184a13.5 13.5 0 0 1-1.727-.11l-4.408 2.883c-.501.265-.678.236-.472-.413l.892-3.678c-2.88-1.46-4.785-3.99-4.785-6.866C1.5 6.665 6.201 3 12 3zm5.907 8.06l1.47-1.424a.472.472 0 0 0-.656-.678l-1.928 1.866V9.282a.472.472 0 0 0-.944 0v2.557a.471.471 0 0 0 0 .222V13.5a.472.472 0 0 0 .944 0v-1.363l.427-.413 1.428 2.033a.472.472 0 1 0 .773-.543l-1.514-2.155zm-2.958 1.924h-1.46V9.297a.472.472 0 0 0-.943 0v4.159c0 .26.21.472.471.472h1.932a.472.472 0 1 0 0-.944zm-5.857-1.092l.696-1.707.638 1.707H9.092zm2.523.488l.002-.016a.469.469 0 0 0-.127-.32l-1.046-2.8a.69.69 0 0 0-.627-.474.696.696 0 0 0-.653.447l-1.661 4.075a.472.472 0 0 0 .874.357l.33-.813h2.07l.299.8a.472.472 0 1 0 .884-.33l-.345-.926zM8.293 9.302a.472.472 0 0 0-.471-.472H4.577a.472.472 0 1 0 0 .944h1.16v3.736a.472.472 0 0 0 .944 0V9.774h1.14c.261 0 .472-.212.472-.472z"
              />
            </g>
          </svg>
        </div>
        <div
          class="github"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 496 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
            />
          </svg>
        </div>
      </div>
    </main>
  </div>
</div>
`;
